lib/src/ansi_code.dart

```diff
@@ -0,0 +1,278 @@
+  static const foregroundCyan =
+      const AnsiCode._('cyan', AnsiCodeType.foreground, 36);
+  static const foregroundLightGray =
+      const AnsiCode._('light gray', AnsiCodeType.foreground, 37);
+  static const foregroundDefault =
+      const AnsiCode._('default', AnsiCodeType.foreground, 39);
+  static const foregroundDarkGray =
+      const AnsiCode._('dark gray', AnsiCodeType.foreground, 90);
+  static const foregroundLightRed =
+      const AnsiCode._('light red', AnsiCodeType.foreground, 91);
+  static const foregroundLightGreen =
+      const AnsiCode._('light green', AnsiCodeType.foreground, 92);
+  static const foregroundLightYellow =
+      const AnsiCode._('light yellow', AnsiCodeType.foreground, 93);
+  static const foregroundLightBlue =
+      const AnsiCode._('light blue', AnsiCodeType.foreground, 94);
+  static const foregroundLightMagenta =
+      const AnsiCode._('light magenta', AnsiCodeType.foreground, 95);
+  static const foregroundLightCyan =
+      const AnsiCode._('light cyan', AnsiCodeType.foreground, 96);
+  static const foregroundWhite =
+      const AnsiCode._('white', AnsiCodeType.foreground, 97);
+
+  //
+  // Background values
+  //
+
+  static const backgroundBlack =
+      const AnsiCode._('black', AnsiCodeType.background, 40);
+  static const backgroundRed =
+      const AnsiCode._('red', AnsiCodeType.background, 41);
+  static const backgroundGreen =
+      const AnsiCode._('green', AnsiCodeType.background, 42);
+  static const backgroundYellow =
+      const AnsiCode._('yellow', AnsiCodeType.background, 43);
+  static const backgroundBlue =
+      const AnsiCode._('blue', AnsiCodeType.background, 44);
+  static const backgroundMagenta =
+      const AnsiCode._('magenta', AnsiCodeType.background, 45);
+  static const backgroundCyan =
+      const AnsiCode._('cyan', AnsiCodeType.background, 46);
+  static const backgroundLightGray =
+      const AnsiCode._('light gray', AnsiCodeType.background, 47);
+  static const backgroundDefault =
+      const AnsiCode._('default', AnsiCodeType.background, 49);
+  static const backgroundDarkGray =
+      const AnsiCode._('dark gray', AnsiCodeType.background, 100);
+  static const backgroundLightRed =
+      const AnsiCode._('light red', AnsiCodeType.background, 101);
+  static const backgroundLightGreen =
+      const AnsiCode._('light green', AnsiCodeType.background, 102);
+  static const backgroundLightYellow =
+      const AnsiCode._('light yellow', AnsiCodeType.background, 103);
+  static const backgroundLightBlue =
+      const AnsiCode._('light blue', AnsiCodeType.background, 104);
+  static const backgroundLightMagenta =
+      const AnsiCode._('light magenta', AnsiCodeType.background, 105);
+  static const backgroundLightCyan =
+      const AnsiCode._('light cyan', AnsiCodeType.background, 106);
+  static const backgroundWhite =
+      const AnsiCode._('white', AnsiCodeType.background, 107);
+
+  /// The numeric value associated with this code.
+  final int code;
```

Comment:
Have you considered making `code` a string?

It will make the sorting below act differently (unless you use a specialized compare function), but the only real use of code is to convert it to a string and embed it in another string.